lib/redox/models/model.rb

```diff
@@ -0,0 +1,49 @@
+module Redox
+  module Models
+    class Model
+      KEY = nil
+
+      def initialize(data)
+        @data = data.freeze
+      end
+
+      def valid?
+        return @data.is_a?(Hash) && @data.include?(self.class::KEY)
+      end
+
+      def raw(key: self.class::KEY)
+        if key.nil?
+          return @data
+        else
+          return @data[key]
+        end
+      end
+
+      def map(mapper: {}, data: self.raw)
+        result = {}
+
+        if (true == mapper.is_a?(Hash))
```

Comment:
Totally agree on comment. Was thinking last night that long-term it would make sense to almost look like ActiveRecord but didn't want to go into the rabbit hole right away. If we keep doing Redox stuff I envision a lot of refactoring in here.

The pattern you asked about is remnants of my first coding style class that have just stubbornly stuck around, the theory is that every once and a while (like probably 100k lines of code, lol) you accidentally do (CONST = variable) and it will throw an error instead of erroneously assigning.

As to not just doing `if predicate`, I'm just a verbose coder and it's just baked into my style now so I have to think to override it, I'm trying to match style where it's already there but sometimes I'm just attacking it and revert to my personal preferences.